opt_einsum/parser.py

```diff
@@ -230,28 +251,20 @@ def parse_einsum_input(operands):
 
         output_list = tmp_operands[-1] if len(tmp_operands) else None
         operands = [possibly_convert_to_numpy(x) for x in operand_list]
-        subscripts = ""
-        last = len(subscript_list) - 1
-        for num, sub in enumerate(subscript_list):
-            for s in sub:
-                if s is Ellipsis:
-                    subscripts += "..."
-                elif isinstance(s, int):
-                    subscripts += get_symbol(s)
-                else:
-                    raise TypeError("For this input type lists must contain either int or Ellipsis")
-            if num != last:
-                subscripts += ","
-
+        try:
```

Comment:
Can you break up these try/except and if blocks with newlines between them with a small comment describing why the steps are needed?

Reply:
Sure, I was going to